python-threatexchange/threatexchange/cli/cli_config.py

```diff
@@ -178,15 +168,25 @@ class _SignalExchangeAccessor:
 
     _parent: "CLISettings"
 
-    def get_all(self) -> t.ValuesView[SignalExchangeAPI]:
+    def get_all(self) -> t.ValuesView[TSignalExchangeAPICls]:
         return self._parent._mapping.exchange.api_by_name.values()
 
-    def get_for_collab(
-        self, collab: collab_config.CollaborationConfigBase
-    ) -> SignalExchangeAPI:
-        return self._parent._mapping.exchange.api_by_name[collab.api]
-
-    def __iter__(self) -> t.Iterator[SignalExchangeAPI]:
+    def get_instance_for_collab(
+        self, collab: TCollabConfig
+    ) -> SignalExchangeAPI[
+        TCollabConfig,
+        fetch_state.FetchCheckpointBase,
+        fetch_state.FetchedSignalMetadata,
+        t.Any,
+        t.Any,
+    ]:
+        api_cls = self._parent._mapping.exchange.api_by_name[collab.api]
+        # This is a hard typing challenge, and cast() doesn't seem to work
+        # By lookup by name, we are guaranteed to pick an API with the right
+        # typing signature, but of an unknown class
+        return api_cls.for_collab(collab)  # type: ignore[arg-type]
```

Comment:
ack that CLI will use default args for all `SignalExchangeAPI.for_collab` invocations. 

I think HMA can reasonably start out with that. We do not need to address this to fix trunk, can add in 'editablity' later.